sdks/python/apache_beam/io/requestresponseio_it_test.py

```diff
@@ -0,0 +1,168 @@
+from apache_beam.options.pipeline_options import PipelineOptions
+
+_HTTP_PATH = "/v1/echo"
+_PAYLOAD = base64.b64encode(bytes('payload', 'utf-8'))
+
+
+class EchoITOptions(PipelineOptions):
+  """Shared options for running integration tests on a deployed
+      ``EchoServiceGrpc`` See https://github.com/apache/beam/tree/master/.test
+      -infra/mock-apis#integration for details on how to acquire values
+      required by ``EchoITOptions``.
+      """
+  @classmethod
+  def _add_argparse_args(cls, parser) -> None:
+    parser.add_argument(
+        '--httpEndpointAddress',
+        required=True,
+        dest='http_endpoint_address',
+        help='The HTTP address of the Echo API endpoint; must being with '
+        'http(s)://')
+    parser.add_argument(
+        '--neverExceedQuotaId',
+        default='echo-should-never-exceed-quota',
+        dest='never_exceed_quota_id',
+        help='The ID for an allocated quota that should never exceed.')
+    parser.add_argument(
+        '--shouldExceedQuotaId',
+        default='echo-should-exceed-quota',
+        dest='should_exceed_quota_id',
+        help='The ID for an allocated quota that should exceed.')
+
+
+# TODO(riteshghorse,damondouglas) replace Echo(Request|Response) with proto
+#   generated classes from .test-infra/mock-apis:
+@dataclass
+class EchoRequest:
+  id: str
+  payload: bytes
+
+
+@dataclass
+class EchoResponse:
+  id: str
+  payload: bytes
+
+
+class EchoHTTPCaller(Caller):
+  """Implements ``Caller`` to call the ``EchoServiceGrpc``'s HTTP handler.
+    The purpose of ``EchoHTTPCaller`` is to support integration tests.
+    """
+  def __init__(self, url: str):
+    self.url = url + _HTTP_PATH
+
+  def call(self, request: EchoRequest) -> EchoResponse:
+    """Overrides ``Caller``'s call method invoking the
+        ``EchoServiceGrpc``'s HTTP handler with an ``EchoRequest``, returning
+        either a successful ``EchoResponse`` or throwing either a
+        ``UserCodeExecutionException``, ``UserCodeTimeoutException``,
+        or a ``UserCodeQuotaException``.
+        """
+
+    try:
+      resp = urllib3.request(
+          "POST",
+          self.url,
+          json={
+              "id": request.id, "payload": str(request.payload, 'utf-8')
+          },
+          retries=False)
+
+      if resp.status < 300:
+        resp_body = resp.json()
+        resp_id = resp_body['id']
+        payload = resp_body['payload']
+        return EchoResponse(id=resp_id, payload=bytes(payload, 'utf-8'))
+
+      if resp.status == 429:  # Too Many Requests
+        raise UserCodeQuotaException(resp.reason)
+
+      raise UserCodeExecutionException(resp.reason)
+
+    except urllib3.exceptions.HTTPError as e:
+      raise UserCodeExecutionException(e)
+
+
+class EchoHTTPCallerTestIT(unittest.TestCase):
+  options: Union[EchoITOptions, None] = None
+  client: Union[EchoHTTPCaller, None] = None
+
+  @classmethod
+  def setUpClass(cls) -> None:
+    cls.options = EchoITOptions()
```

Comment:
since `httpEndpointAddress` is a required argument, this will throw error. This is causing a lot of test suites to fail.

Reply:
we need to pass `httpEndpointAddress` to `EchoITOptions()` here

Reply:
Thank you for reviewing :-). I was confused about the convention for integration tests in the Python SDK. To be on the safe side, should I just remove the requirement flag and check for it when needed in the test?

Reply:
Resolving as I removed the `required=True` from the option.